app/incident/incident.py

```diff
@@ -75,6 +80,24 @@ def generate_chain(self, chain=None):
                 self.chain_put(index=index, datetime_=dt, type_=type_, identifier=value)
         self.dump()
 
+    def _unchain(self, chains, steps):
+        if any('chain' in step for step in steps):
+            steps_result = list()
+            for step in steps:
+                type_, value = next(iter(step.items()))
+                if type_ == 'chain':
+                    nested_chain = chains.get(value)
+                    if nested_chain is None:
+                        logger.warning(f"Chain '{value}' not found. Check impulse.yml")
+                        continue
+                    nested_steps = nested_chain.steps
+                    steps_result.extend(self._unchain(chains, nested_steps))
+                else:
+                    steps_result.append({type_: value})
+            return steps_result
+        else:
+            return steps
+
```

Comment:
Some readablility and code quality improvements. It'd be good approach to add docstrings everywhere.

```suggestion
    def _unchain(self, chains, steps):
        # If no chains to resolve, return steps as is
        if not any('chain' in step for step in steps):
            return steps

        resolved_steps = []
        for step in steps:
            type_, value = next(iter(step.items()))

            if type_ == 'chain':
                nested_chain = chains.get(value)
                if nested_chain is None:
                    logger.warning(f"Chain '{value}' not found. Check impulse.yml")
                    continue

                # Recursively resolve nested chain steps
                nested_steps = nested_chain.steps
                resolved_steps.extend(self._unchain(chains, nested_steps))
            else:
                resolved_steps.append({type_: value})

        return resolved_steps
```